### PR TITLE
AP_NavEKF3: accept set origin even when using GPS

### PR DESCRIPTION
**Pull request by rmackay9**
This removes two unnecessary constraints on when the EKF origin can be set and adds an autotest (courtesy of PeterB).

The existing constraints cause problems when the vehicle is simultaneously using GPS and OpticalFlow (e.g. EK3_SRC1_POSXY = 3/GPS, EK3_SRC1_VELXY=5/OptFlow, EK3_SRC_OPTIONS = 1/FuseAllVelocities) and the vehicle's start location is indoors where there is no GPS.  In this situation the user is currently stuck because the constraint means the origin must come from the GPS but the GPS cannot provide a Location.  I've received two reports of this inconvenience ([the most recent report is here](https://discuss.ardupilot.org/t/lua-scripting-set-ekf-origin/117935)).

Note that I've also removed a constraint for, "PV_AidingMode == AID_ABSOLUTE".  As far as I know there is no way the EKF can be in AID_ABSOLUTE without an origin which we check for moments later in setOrigin().  It does no harm but it also does no good and is probably just a left-over from the earlier changes related to a common origin or the moving internal origin.

I have tested this in SITL to confirm that it resolves the issue.  Using master/latest with both GPS and optical flow enabled and SIM_GPS_DISABLE=1 the EKF origin could not be set using MAVProxy's right-mouse-button menu.
image

With this PR the origin could be set.
image

In addition the vehicle could be force armed (to bypass the "GPS 1: Bad fix" message) and flown.  Below is a screen shot of a test in which the GPS was disabled and the EKF origin was purposely set 100m off from the correct Location.  The vehicle was armed and flew correctly albeit with a persistent position offset.
image

When GPS was restored (e.g. param SIM_GPS_DISABLE 0) the vehicle correctly (and smoothly) jumped back to its actual position).
image

Note that I found a [bug in master](https://github.com/ArduPilot/ardupilot/issues/27103) during testing.  This PR makes it no better but also no worse.

### CODE REVIEW

**Review by peterbarker — COMMENTED** [review]
This sanity check was in place in case the user managed to se origin a very long way away from where the vehicle actually was.

We believe that that isn't a problem post-Extreme ArduPilot moving-origin stuff.

What testing have you done for the case which Extreme ArduPilot *thinks* it fixed?

I've created this PR which was my attempt at testing this scenario: https://github.com/ArduPilot/ardupilot/pull/27084 in my branch equivalent to this one.

**Comment by peterbarker** [comment]
... in fact, here's my pre-existing PR: https://github.com/ArduPilot/ardupilot/pull/25110

Note tridge's objection on my PR which would also seem to be the same objection on your PR here (you're also not modifying the function tridge pointed to)

**Comment by rmackay9** [comment]
Hi @peterbarker,

Actually you *have* addressed Tridge's issue by removing the similar checks in NavEKF3_core::setOriginLLH(). This PR also removes those lines.

File: libraries/AP_NavEKF3/AP_NavEKF3.cpp [hunk 1]
```diff
@@ -1392,11 +1392,9 @@ bool NavEKF3::setOriginLLH(const Location &loc)
...
     }
-    if ((sources.getPosXYSource() == AP_NavEKF_Source::SourceXY::GPS) || common_origin_valid) {
-        // we don't allow setting of the EKF origin if using GPS
-        // or if the EKF origin has already been set.
-        // This is to prevent accidental setting of EKF origin with an
-        // invalid position or height or causing upsets from a shifting origin.
+    if (common_origin_valid) {
+        // we don't allow setting the EKF origin if it has already been set
+        // this is to prevent causing upsets from a shifting origin.
         GCS_SEND_TEXT(MAV_SEVERITY_WARNING, "EKF3 refusing set origin");
```
[the remark this answers is not in the document]
Comment (rmackay9) [reply libraries/AP_NavEKF3/AP_NavEKF3.cpp:1398]: I agree and this also crossed my mind actually.

**Review by rmackay9 — COMMENTED** [review]
LGTM.

Strangely GitHub is not allowing me to approve this PR but it looks good!

**Review by tridge — APPROVED** [review]

**Pull request merged**